python/delta_sharing/rest_client.py

```diff
@@ -219,6 +224,14 @@ def query_table_metadata(self, table: Table) -> QueryTableMetadataResponse:
                 metadata=Metadata.from_json(metadata_json["metaData"]),
             )
 
+    @retry_with_exponential_backoff
+    def query_table_version(self, table: Table) -> QueryTableVersionResponse:
+        headers = self._head_internal(
+            f"/shares/{table.share}/schemas/{table.schema}/tables/{table.name}"
+        )
+        table_version = int(headers.get("delta-table-version", None) or -1)
```

Comment:
Could you raise an error when the header is missing? This is a required header defined in the protocol. If a server doesn't return it, it should be a bug in the server and it's better to raise an error.

Reply:
if the table does not exist on the server, should the server still return a delta-table-version in the header? Happy to raise a LookupError when the header is missing

Reply:
Good point. This reminds me that we are missing `response.raise_for_status()` in `_head_internal`. We should fail if the server turns non 200 response. Could you also add a test for this?